Assets/Scripts/UI/MintDeckButtonHandler.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+[RequireComponent(typeof(Button))]
+public class MintDeckButtonHandler : MonoBehaviour
+{
+    void Start()
+    {
+        GetComponent<Button>().onClick.AddListener(OnButtonClicked);
+    }
+
+    async void OnButtonClicked()
+    {
+        GetComponent<Button>().interactable = false;
+        var tx = await PepemonCardDeck.CreateDeck();
+        // TODO: show error label on failure
```

Comment:
Agreed, it would be good to have a "current transaction" popup box that we could show the status on (e.g. waiting, finished, reverted, etc.). We can add this as another task.